content/news/2023-08-14-chatGPT/index.md

```diff
@@ -0,0 +1,55 @@
+---
+title: "ChatGPT in Galaxy"
+date: "2023-08-14"
+authors: Anup Kumar
+tease: "Use ChatGPT in Galaxy Europe via JupyterLab interactive tool."
+hide_tease: true
+subsites: [all-eu,global]
+---
+
+### Generative AI (ChatGPT) as a programming assitant
+
+Many generative AI models by [OpenAI](https://openai.com/) are now accessible via a [UI plugin](https://github.com/jupyterlab/jupyter-ai) in the GPU-enabled JupyterLab interactive tool on Galaxy Europe. These AI models from OpenAI have been trained on an enormous amount of text that provides reasonable answers to queries - from literature to programming. In several ways, models such as ChatGPT can assist programmers in generating a backbone of general-purpose code blocks and methods, finding errors in code or developing explanations of already written code or existing errors, and for many different purposes. 
+
+### Usage
+
+![Jupyternaut 3](./chatgpt_3.png) 
+
+By clicking on "chat" icon in the (bottom) left panel of JupyterLab, "Jupyternaut" programming assistant can be configured. It asks to choose a "language model" (an LLM). (Optionally) Further, an "embedding" model can be selected to make the assistant learn about the local file structures - files inside JupyterLab. User must provide their OpenAI access token to get responses to their queries. These tokens are for personal use and are NOT saved inside Galaxy. Refer to the above image. 
+
+The Jupyternaut panel provides a chatbox for writing questions. When a query is submitted, an API call is made to the chosen OpenAI model and responses are shown in the same panel. Refer to the following two images.
+
+![Jupyternaut 1](./chatgpt_1.png)
+
+Follow-up queries can also be made: 
+
+
+![Jupyternaut 2](./chatgpt_2.png)
+
+Another way to use ChatGPT is via a notebook using "magic commands `%%ai`". Refer to the following image.
+
+![ChatGPT in a notebook](./chatgpt_4.png)
+
+
+### Pricing for OpenAI token
+
+The cost of using OpenAI models depends on factors such as the type of model (LLM) used and the number of tokens. The [price chart](https://openai.com/pricing) describes the detailed cost of using different LLMs. Models such as GPT4 are more expensive than GPT-3.5 Turbo. Usage cost also depends on the number of tokens/words used in the query and its response. Refer to the [tokens guide](https://platform.openai.com/docs/guides/gpt/managing-tokens) for more understanding. A straightforward example of how tokens are managed inside models is - the string "ChatGPT is great!" is encoded into six tokens: ["Chat", "G", "PT", " is", " great", "!"]. Using the GPT-3.5 Turbo model for 1000 tokens costs $0.0015.
+
+
+### How to access on Galaxy Europe:
```

Comment:
Isn't that section a bit redundant with the general usage section?

Reply:
@bgruening, thanks a lot for your comments. I have update the blog according to your comments